### PR TITLE
Fix planning save bug and Sprint state mgmt

### LINKED ISSUE #1: Planning Save Bug
The planning documents for a team in a Sprint, for some reason, are duplicating/mirroring each other's state from one week to the next. I think their state in the Redux store isn't probably being cleared, week to week perhaps?

To reproduce: create a sprint, write out the goals for that sprint. Then, create a new sprint - and go to the planning. The planning text from the last weeks sprint has carried over for some reason.

### LINKED ISSUE #157: Refactor: Sprint component is very long and could benefit from refactoring
## Description
The component `arena/Sprint.js` has a whopping 556 lines of code! Looking through, a good amount of it appears to be repetitive code that could be refactored or broken out into subcomponents.

## Goals
- [ ] Reduce the total lines of code
- [ ] Reduce the amount of repetitive code
- [ ] Improve code efficiency and readability
- [ ] Simplify state management for states that are always all updated at the same time
- [ ] Add comments as necessary to help orient new contributors navigating the file for the first time.

### PR DESCRIPTION
**Pull request by jayeclark**
Pull-Request for `paretOS`

## Description
- Fixes the planning forms mirroring bug
- Simplifies state management in Sprint by removing consts from state (it is not possible for the sprint id, user id, or start date of a sprint to change unless a different URL is accessed, in which case the Sprint component would remount, so there is no reason to store them in state - they are available immediately when the component is mounted due to being stored in redux.)
- Fixes MUI custom variant typing issue

## Relates to
Fixes #1 
Fixes #157 remaining checkbox re: state management

## Reviewers
- @mikhael28

### CODE REVIEW

**Comment by jayeclark** [comment]
Double checking on a planning save/websocket bug before I move out of draft.

**Comment by mikhael28** [comment]
Sorry for the delay, going to pull this down and review tomorrow! If you have time before then, can you resolve merge conflicts with this branch? ty ty, excited to close Issue #1!

**Comment by jayeclark** [comment]
Done!

File: src/arena/Sprint.js [hunk 1]
```diff
@@ -34,27 +35,67 @@ import ArenaDynamicForms from "./ArenaDynamicForms";
  */
 
 function Sprint(props) {
-  const theme = useTheme();
-  const [status, setStatus] = useState("early");
-  const [activeSprintId, setActiveSprintId] = useState(0);
-  const [startDate, setStartDate] = useState(0);
-  const [displayDay, setDisplayDay] = useState(0);
+  // Identify the sprint index
+  const location = useLocation();
```
Comment (mikhael28) [inline src/arena/Sprint.js:39]: Very cool, I didn't know this hook existed!

**Review by mikhael28 — APPROVED** [review]
Great work! Pulled this down and tested it myself - had some weird issue on Safari publishing multiple web socket updates, but it works just fine in Chrome. Will revisit that later, need to do an in-depth Safari/Firefox/Opera Mini audit either way. Merging - issue #1 is closed!!!!!

**Pull request merged**